src/perf/PerfListener.cc

```diff
@@ -42,7 +42,10 @@ void PerfListener::onPreFork() {
     TRACE();
 
     barrier_ =
-            withErrnoCheck("mmap shared memory", mmap,
+            withGuardedErrnoCheck(
+                "mmap shared memory",
+                [](auto&& result) -> bool { return result != MAP_FAILED; },
```

Comment:
To robiło buga tylko na i386, czy na amd64 też?
Also, IMO osobny commit.

Reply:
IIRC przez przypadek tylko na i386. Chyba chodziło o coś takiego, że ta poprzednia wersja withErrnoCheck to castowała na liczbę ze znakiem i na i386 czasami dostawaliśmy poprawne adresy mniejsze od zera.

Reply:
Dziwne że na amd64 nigdy takich nie dostaliśmy.